### PR TITLE
New: Review schema block.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

Added a new review schema block.

### Types of changes

New feature (non-breaking change which adds functionality)

### How has this been tested?

- Drag and drop review schema block.
- Add schema related attribute and check it on the schema tool.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/review/edit.js [hunk 1]
```diff
@@ -0,0 +1,1091 @@
...
+
+class UAGBRatingEdit extends Component {
+
+	constructor() {
+
+		super( ...arguments )
+
+		this.onRemoveImage = this.onRemoveImage.bind( this )
+		this.onSelectImage = this.onSelectImage.bind( this )
+		// this.onUpdateDate  = this.onUpdateDate.blind( this )
```
Comment (RajkiranBagal) [inline src/blocks/review/edit.js:78]: Remove commented code

**Comment by RajkiranBagal** [comment]
@AbhijitNage123 Please check these Console errors - https://a.cl.ly/X6u00x7B

**Comment by RajkiranBagal** [comment]
Verify this from Anjali Mam - https://a.cl.ly/eDuwwQXZ

File: src/blocks/review/edit.js [hunk 2]
```diff
@@ -0,0 +1,1091 @@
...
+} = wp.blockEditor
+
+const {
+	PanelBody,
+	RangeControl,
+	SelectControl,
+	ToggleControl,
+	Button,
+	TextControl,
+	DatePicker,
```
Comment (RajkiranBagal) [inline src/blocks/review/edit.js:37]: Remove unused package

File: src/blocks/review/edit.js [hunk 3]
```diff
@@ -0,0 +1,1091 @@
...
+				inactiveStarColor,
+				activeStarColor,
+				sku,
+				identifier,
+				identifierType,
+				offerType,
+				offerCurrency,
+				offerStatus,
+				offerPrice,
+				// offerCount,
```
Comment (RajkiranBagal) [inline src/blocks/review/edit.js:193]: Remove unwanted attrs like this.

File: src/blocks/review/edit.js [hunk 4]
```diff
@@ -0,0 +1,1091 @@
...
+			/>{ loadHeadingGoogleFonts }
+			{ loadSubHeadingGoogleFonts }
+			{ loadContentGoogleFonts }</div>
+		];
+	}} 
+	export default compose(
+		withSelect( ( select, ownProps ) => {
+			const newAverage = ownProps.attributes.parts.map((i) => i.value).reduce((total, v) => total + v) / ownProps.attributes.parts.length;
+			
+				var offers = {}
```
Comment (RajkiranBagal) [inline src/blocks/review/edit.js:980]: Remove unused declarations

File: src/blocks/review/schema-notices.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+        var emptyItems = [];
+
+        switch ( itemType ) {
+            case "Product":
+                if ( 'undefined' === typeof sku || '' === sku ) {
+                    emptyItems.push( 'Mising SKU' );
+                }
+                if ( 'undefined' === typeof brand || '' === brand ) {
+                    emptyItems.push( 'Mising Brand' );
```
Comment (RajkiranBagal) [inline src/blocks/review/schema-notices.js:51]: Can we make these notices Translatable?

File: readme.txt [hunk 6]
```diff
@@ -70,6 +70,7 @@ We also have an informative and detailed article that will help you to <a href="
 <strong>An Ultimate Gutenberg Blocks Library</strong>
 
 <ul>
+	<li><a href="" target="_blank">Review block </a>: With this block, you can easily display review schema on post/pages. Further, you can style it with colors, typography.</li>
```
Comment (RajkiranBagal) [inline readme.txt:73]: Verify this from Anjali Mam

File: readme.txt [hunk 7]
```diff
@@ -161,6 +162,7 @@ When you use the Ultimate Addons for Gutenberg along with the free Astra theme,
 == Changelog ==
 
 = 1.19.0 =
+* New: Review block.
```
Comment (RajkiranBagal) [inline readme.txt:165]: Shall the name be `Review Schema Block` ?

Comment (AbhijitNage123) [reply readme.txt]: It should be `Review Schema Block`. Updated

File: src/blocks/review/edit.js (hunk 4 above)
Comment (AbhijitNage123) [reply src/blocks/review/edit.js:974]: `newAverage` used in the schema for a `ratingValue`

Comment (RajkiranBagal) [reply src/blocks/review/edit.js:974]: I am talking about these declarations

`var offers = {}
				var itemReviewed = {}`

Comment (AbhijitNage123) [reply src/blocks/review/edit.js]: Removed

**Comment by AbhijitNage123** [comment]
> @AbhijitNage123 Please check these Console errors - https://a.cl.ly/X6u00x7B

I'm not able to reproduce this at my end. Can you please share the test cases for this?

File: readme.txt (hunk 6 above)
Comment (AbhijitNage123) [reply readme.txt]: Done

**Comment by AbhijitNage123** [comment]
> Verify this from Anjali Mam - https://a.cl.ly/eDuwwQXZ

Updated

File: src/blocks/review/style.scss [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+			padding-top:10px;
+			padding-bottom:10px;
+		}
+	  .uagb_review_entry {
+		display: flex;
+		padding: 10px 0 10px 0;
+		border-bottom: 1px solid #e9e9e9;
+	  }
+	  .uagb_review_summary_title {
+		margin: 23px 0 13px 0 !important;
```
Comment (vrundakansara) [inline src/blocks/review/style.scss:17]: Any reason why we have added `!important`

File: src/blocks/review/style.scss [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+		font-size: 24px !important;
+		font-weight: 600 !important;
+	  }
+	  .uagb_review_summary .uagb_review_overall_value>.rich-text {
```
Comment (vrundakansara) [inline src/blocks/review/style.scss:21]: Do we need rich text css here?

File: src/blocks/review/style.scss (hunk 8 above)
Comment (AbhijitNage123) [reply src/blocks/review/style.scss]: Removed.

**Review by vrundakansara — APPROVED** [review]

**Pull request merged**